virttest/qemu_devices/qdevices.py

```diff
@@ -246,7 +248,18 @@ def get_children(self):
 
     def cmdline(self):
         """ :return: cmdline command to define this device """
-        raise NotImplementedError
+        if self.cmdline_format == "json":
+            return self._cmdline_json()
+        else:
+            return self._cmdline_raw()
```

Comment:
do we want to check the exact string 'raw' here in else, and raise a ValueError exception in other cases?
then if there is a typo, e.g. 'jons', i.e. if we want json format, but give a typo, the cmd will be in raw format.
But, as the format is defined in templates, and it seems there is no test scenario to cover the format test, I think it's OK too.